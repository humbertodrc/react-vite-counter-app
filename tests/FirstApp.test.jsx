import { render } from '@testing-library/react'
import { FirstApp } from '../src/FirstApp'

describe('Pruebas en <FirstApp />', () => {

  const title = 'Hola soy un Titulo';


  test('Debe hacer match con el snapshot', () => {

    const {container} = render(<FirstApp title={title} />);
    
    expect(container).toMatchSnapshot();

  })

  test('Debe mostar el titulo en un h1', () => {
    
    const { container, getByText } = render(<FirstApp title={title} />);
    
    expect(getByText(title)).toBeTruthy();

    // Buscar si existe un h1 con JS
    const h1 = container.querySelector('h1');
    expect(h1.innerHTML).toContain(title);
  
  })

})