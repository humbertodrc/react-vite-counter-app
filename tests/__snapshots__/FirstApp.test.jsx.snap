// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola soy un Titulo
  </h1>
  <p>
    No hay subt√≠tulo
  </p>
  <p>
    Humberto Rivero
  </p>
</div>
`;
